debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -78,6 +79,9 @@ public PostgresStreamingChangeEventSource(PostgresConnectorConfig connectorConfi
         this.clock = clock;
         this.schema = schema;
         this.offsetContext = (offsetContext != null) ? offsetContext : PostgresOffsetContext.initialContext(connectorConfig, connection, clock);
+        // replication slot could exist at the time of starting Debezium so we will stream from the position in the slot
+        // instead of the last position in the database
+        this.hasStartLsnStoredInContext = (offsetContext != null);
```

Comment:
This looks good to me. I had hoped you could instead create an initial empty context here, but my attempt to do so caused test failures.